extensions/pkg/webhook/utils.go

```diff
@@ -65,70 +67,91 @@ func SerializeCommandLine(command []string, n int, sep string) string {
-		return &pvcs[i]
+	for _, pvc := range pvcs {
+		if pvc.Name == name {
+			return &pvc
+		}
 	}
 	return nil
 }
 
 // UnitWithName returns the unit with the given name if it exists in the given slice, nil otherwise.
 func UnitWithName(units []extensionsv1alpha1.Unit, name string) *extensionsv1alpha1.Unit {
-	if i := unitWithNameIndex(units, name); i >= 0 {
-		return &units[i]
+	for _, unit := range units {
+		if unit.Name == name {
+			return &unit
+		}
 	}
 	return nil
 }
 
 // FileWithPath returns the file with the given path if it exists in the given slice, nil otherwise.
 func FileWithPath(files []extensionsv1alpha1.File, path string) *extensionsv1alpha1.File {
-	if i := fileWithPathIndex(files, path); i >= 0 {
-		return &files[i]
+	for _, file := range files {
+		if file.Path == path {
+			return &file
+		}
 	}
 	return nil
 }
 
 // UnitOptionWithSectionAndName returns the unit option with the given section and name if it exists in the given slice, nil otherwise.
 func UnitOptionWithSectionAndName(opts []*unit.UnitOption, section, name string) *unit.UnitOption {
-	if i := unitOptionWithSectionAndNameIndex(opts, section, name); i >= 0 {
-		return opts[i]
+	for _, opt := range opts {
+		if opt.Section == section && opt.Name == name {
+			return opt
+		}
 	}
+
 	return nil
 }
 
 // EnsureStringWithPrefix ensures that a string having the given prefix exists in the given slice
-// with a value equal to prefix + value.
+// and all matches are with a value equal to prefix + value.
 func EnsureStringWithPrefix(items []string, prefix, value string) []string {
-	item := prefix + value
 	if i := StringWithPrefixIndex(items, prefix); i < 0 {
-		items = append(items, item)
-	} else if items[i] != item {
-		items = append(append(items[:i], item), items[i+1:]...)
+		return append(items, prefix+value)
+	}
+
+	for i, item := range items {
+		if !strings.HasPrefix(item, prefix) {
+			continue
+		}
+		if item != prefix+value {
+			items = append(append(items[:i], prefix+value), items[i+1:]...)
+		}
 	}
 	return items
 }
 
 // EnsureNoStringWithPrefix ensures that a string having the given prefix does not exist in the given slice.
 func EnsureNoStringWithPrefix(items []string, prefix string) []string {
-	if i := StringWithPrefixIndex(items, prefix); i >= 0 {
-		items = append(items[:i], items[i+1:]...)
-	}
-	return items
+	return slices.DeleteFunc(items, func(s string) bool {
+		return strings.HasPrefix(s, prefix)
+	})
 }
 
 // EnsureStringWithPrefixContains ensures that a string having the given prefix exists in the given slice
-// and contains the given value in a list separated by sep.
+// and all matches contain the given value in a list separated by sep.
 func EnsureStringWithPrefixContains(items []string, prefix, value, sep string) []string {
 	if i := StringWithPrefixIndex(items, prefix); i < 0 {
-		items = append(items, prefix+value)
-	} else {
+		return append(items, prefix+value)
+	}
+
+	for i, item := range items {
```

Comment:
We can use `items[i]` instad of `items = append(append(items[:i], prefix+strings.Join(values, sep)), items[i+1:]...)`